XenOvfTransport/Export.cs

```diff
@@ -354,6 +355,37 @@ private EnvelopeType _export(Session xenSession, string targetPath, string ovfna
                         OVF.AddOtherSystemSettingData(ovfEnv, vsId, "vgpu", sb.ToString(), OVF.GetContentMessage("OTHER_SYSTEM_SETTING_DESCRIPTION_4"));
                     }
                 }
+
+                var allProxies = PVS_proxy.get_all_records(xenSession).Select(kvp => kvp.Value).ToList();
+                
+                string pvsSiteUuid = string.Empty;
+
+                foreach (var p in allProxies.Where(p => p != null && p.VIF != null))
+                {
+                    var vif = xenSession.Connection.Resolve(p.VIF);
+                    if (vif != null)
+                    {
+                        var vmFromVif = xenSession.Connection.Resolve(vif.VM);
+                        if (vmFromVif != null && vmFromVif.uuid == vm.uuid)
+                        {
+                            var pvsSite = xenSession.Connection.Resolve(p.site);
+                            if (pvsSite != null)
+                            {
+                                pvsSiteUuid = pvsSite.uuid;
+                                
+                                break;

```

Comment:
Currently it exits the loop once it's found a pvs proxy object that refers the current vm and has a valid pvs site assigned to it. However, I see that doing so makes the loop to continue in case it finds a pvs proxy object for the vm, but can't resolve the referred pvs site. I can't see significant difference, but as we discussed it makes sense to assume the first object is fine and not to search for a next one in the not-expected case should the pvs site be not resolvable. So moving the break statement outside.